library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -61,25 +70,48 @@ public void setup(PortContext context)
     }
   };
 
-  @OutputPortFieldAnnotation(optional = true)
   public transient DefaultOutputPort<INPUT> err = new DefaultOutputPort<INPUT>();
```

Comment:
Why don't we remove the error port and add process error tuples method which by default logs error messages. I think logging is the obvious default behavior.

Users can add error port and  override this processErrorTuples().  The reason is that we don't know what kind of tuples are error tuples - same type as Input tuples or different.  
From the customer use cases seen so far they are either input tuples which are logged to files or the type is completely different.
Currently we don't have a reason to assume that error tuples are KeyValPair so that shouldn't be picked as default.